### PR TITLE
fix for #101: synchronize body map in DynamicMockProvider (#102)

### PR DESCRIPTION
**Pull request by speekha**
* fix for #101: synchronize body map in DynamicMockProvider

### CODE REVIEW

**Comment by speekha** [comment]
@lsuski I've pushed my fix. Behavior should not have changed and race condition is no longer present.
Let me know what you think before I merge the PR.

File: tests/src/test/java/fr/speekha/httpmocker/interceptor/JavaApiTest.java [hunk 1]
```diff
@@ -18,11 +18,7 @@
 
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.Nested;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
```
Comment (speekha) [inline tests/src/test/java/fr/speekha/httpmocker/interceptor/JavaApiTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'org.junit.jupiter.api'](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834)

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834&bid=20676250)

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834&bid=20676250)

File: tests/src/test/java/fr/speekha/httpmocker/interceptor/JavaApiTest.java (hunk 1 above)
Comment (speekha) [inline tests/src/test/java/fr/speekha/httpmocker/interceptor/JavaApiTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'org.junit.jupiter.api'](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834)

**Comment by lsuski** [comment]
It looks simplier and there is no state in DynamicMockProvider. However I'm not sure if removing test for simultenous requests is a good idea. Maybe `repeat(1000)` is not needed anymore but simple test ensuring that requests do not interfere could remain.

**Comment by speekha** [comment]
@lsuski I did hesitate before removing it. The only reason to add that test in the first place was my faulty implementation. 
I wasn't very happy with 1000 threads being started in the tests. Maybe I'll keep it for now and refactor it later.

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834&bid=20676250)

File: tests/src/test/java/fr/speekha/httpmocker/interceptor/JavaApiTest.java (hunk 1 above)
Comment (speekha) [inline tests/src/test/java/fr/speekha/httpmocker/interceptor/JavaApiTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'org.junit.jupiter.api'](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834)

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834&bid=20676250)

File: tests/src/test/java/fr/speekha/httpmocker/interceptor/JavaApiTest.java (hunk 1 above)
Comment (speekha) [inline tests/src/test/java/fr/speekha/httpmocker/interceptor/JavaApiTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'org.junit.jupiter.api'](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834)

Comment (speekha) [inline tests/src/test/java/fr/speekha/httpmocker/interceptor/JavaApiTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'org.junit.jupiter.api'](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834)

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834&bid=20676250)

**Comment by lsuski** [comment]
So maybe just remove repeat(1000) and invoke requests once

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834&bid=20676250)

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834&bid=20676250)

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 21
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834&bid=20676250)

**Comment by speekha** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 21
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/speekha/httpmocker/pullRequest?prid=6371834&bid=20676250)

**Pull request merged**